Libraries/LibWeb/HTML/HTMLImageElement.cpp

```diff
@@ -171,22 +172,40 @@ bool HTMLImageElement::is_image_available() const
 
 Optional<CSSPixels> HTMLImageElement::intrinsic_width() const
 {
-    if (auto image_data = m_current_request->image_data())
-        return image_data->intrinsic_width();
+    if (auto image_data = m_current_request->image_data()) {
+        auto const image_orientation = computed_properties()
+            ? to_gfx_image_orientation(computed_properties()->image_orientation())
+            : Gfx::ImageOrientation::FromExif;
+
+        return image_data->intrinsic_width(image_orientation);
+    }
+
     return {};
 }
 
 Optional<CSSPixels> HTMLImageElement::intrinsic_height() const
 {
-    if (auto image_data = m_current_request->image_data())
-        return image_data->intrinsic_height();
+    if (auto image_data = m_current_request->image_data()) {
+        auto const image_orientation = computed_properties()
+            ? to_gfx_image_orientation(computed_properties()->image_orientation())
+            : Gfx::ImageOrientation::FromExif;
```

Comment:
You have to do this a lot. Is it worth storing the Gfx::ImageOrientation in the computed properties instead of the CSS::ImageOrientation?